nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/ChangeMasterPasswordCommandDAS.java

```diff
@@ -120,11 +127,29 @@ protected int executeCommand() throws CommandException {
             domainConfig.put(DomainConfig.K_SAVE_MASTER_PASSWORD, savemp);
             manager.changeMasterPassword(domainConfig);
 
+            try {
+                if (dataGridEncryptionEnabled()) {
+                    logger.warning("Data grid encryption is enabled - " +
+                            "you will need to regenerate the encryption key");
+                }
+            } catch (IOException | SAXException | ParserConfigurationException | NullPointerException exception) {
+                logger.warning("Could not determine if data grid encryption is enabled - " +
+                        "you will need to regenerate the encryption key if it is");
+            }
+
             return 0;
         } catch(Exception e) {
             throw new CommandException(e.getMessage(),e);
         }
     }
+
+    private boolean dataGridEncryptionEnabled() throws IOException, SAXException, ParserConfigurationException {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document document = builder.parse(getDomainXml());
+        return Boolean.valueOf(document.getElementsByTagName("hazelcast-runtime-configuration")
+                .item(0).getAttributes().getNamedItem("datagrid-encryption-enabled").getNodeValue());
+    }
```

Comment:
Yep, it's horrendously heavyweight: if you have any suggestions I'd love them.

I don't have access to config beans from the CLI as the server isn't running.
I looked at the MiniXmlParser class but that's specific to configs, not root level domain stuff.
I _could_ write my own XmlStreamParser as the hazelcast configuration _tends_ to be at the top, but as the **many** comments in the other parsers state, you can't guarantee the order.
Probably still quicker though, I'll have another look

Reply:
Only suggestion I have is to leave a comment on the method explaining why this has to be done this way, like "Server isn't running, no access to config beans => have to do it manually" so that later "us" seeing this don't go "ouch, does this have to be like that?" again.